src/policy/policy.cpp

```diff
@@ -209,9 +215,15 @@ bool AreInputsStandard(const CTransaction& tx, const CCoinsViewCache& mapInputs)
             if (stack.empty())
                 return false;
             CScript subscript(stack.back().begin(), stack.back().end());
-            if (subscript.GetSigOpCount(true) > MAX_P2SH_SIGOPS) {
+            const auto p2sh_sigops{subscript.GetSigOpCount(true)};
+            if (p2sh_sigops > MAX_P2SH_SIGOPS) {
                 return false;
             }
+            sigops += p2sh_sigops;
+        }
+
+        if (sigops > MAX_LEGACY_SIGOPS) {
```

Comment:
Although there are no `break` or `continue` statements in the loop, it would feel a bit more robust to move this check below the loop.

Reply:
I understand the slight additional complexity concern but for a transaction submitted to us with no PoW i think we should discard it as soon as we know we won't accept it. I really don't think we should keep doing unnecessary work by iterating, solving scripts and counting.